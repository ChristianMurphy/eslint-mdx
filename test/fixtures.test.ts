import path from 'path'

import { ESLint } from 'eslint'

const getCli = (lintCodeBlocks = false) =>
  new ESLint({
    ignore: false,
    useEslintrc: false,
    baseConfig: {
      parserOptions: {
        ecmaVersion: 2020,
        sourceType: 'module',
      },
      extends: [
        'plugin:react/recommended',
        'plugin:unicorn/recommended',
        'plugin:prettier/recommended',
        'plugin:mdx/recommended',
      ],
      settings: {
        react: {
          version: 'detect',
        },
      },
      overrides: lintCodeBlocks
        ? [
            {
              files: '**/*.{md,mdx}/**',
              rules: {
                'prettier/prettier': 0,
              },
            },
            {
              files: '*.{md,mdx}',
              // related to https://github.com/eslint/eslint/issues/14207
              rules: {
                'prettier/prettier': 0,
              },
              settings: {
                'mdx/code-blocks': true,
              },
            },
          ]
        : [],
    },
  })

describe('fixtures', () => {
  it('should match all snapshots', async () => {
    const results = await getCli().lintFiles('test/fixtures/*')
    for (const { filePath, messages } of results) {
      expect(messages).toMatchSnapshot(path.basename(filePath))
    }
  })

  describe('lint code blocks', () => {
    it('should work as expected', async () => {
      const results = await getCli(true).lintFiles(
        'test/fixtures/code-blocks.{md,mdx}',
      )
      for (const { filePath, messages } of results) {
        expect(messages).toMatchSnapshot(path.basename(filePath))
      }
    })
  })
})
