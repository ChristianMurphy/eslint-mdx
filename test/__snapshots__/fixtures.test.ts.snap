// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures lint code blocks should work as expected: code-blocks.md 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 1,
    "endLine": 13,
    "line": 10,
    "message": "Filename is not in kebab case. Rename it to \`0-0.js\`.",
    "messageId": "renameToCase",
    "nodeType": "Program",
    "ruleId": "unicorn/filename-case",
    "severity": 2,
  },
  Object {
    "column": 2,
    "endColumn": 2,
    "endLine": 39,
    "line": 36,
    "message": "Filename is not in kebab case. Rename it to \`3-3.md\`.",
    "messageId": "renameToCase",
    "nodeType": "Program",
    "ruleId": "unicorn/filename-case",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 38,
    "fix": Object {
      "range": Array [
        410,
        415,
      ],
      "text": "# abc",
    },
    "line": 38,
    "message": "Do not use headings with similar content (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 38,
    "fix": Object {
      "range": Array [
        410,
        415,
      ],
      "text": "# abc",
    },
    "line": 38,
    "message": "Don’t use multiple top level headings (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 2,
    "endColumn": 2,
    "endLine": 49,
    "line": 44,
    "message": "Filename is not in kebab case. Rename it to \`4-4.mdx\`.",
    "messageId": "renameToCase",
    "nodeType": "Program",
    "ruleId": "unicorn/filename-case",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 48,
    "fix": Object {
      "range": Array [
        476,
        481,
      ],
      "text": "# abc",
    },
    "line": 48,
    "message": "Do not use headings with similar content (3:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 48,
    "fix": Object {
      "range": Array [
        476,
        481,
      ],
      "text": "# abc",
    },
    "line": 48,
    "message": "Don’t use multiple top level headings (3:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        7,
        12,
      ],
      "text": "# abc",
    },
    "line": 3,
    "message": "Do not use headings with similar content (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        7,
        12,
      ],
      "text": "# abc",
    },
    "line": 3,
    "message": "Don’t use multiple top level headings (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 1,
    "endColumn": 4,
    "endLine": 17,
    "fix": Object {
      "range": Array [
        158,
        187,
      ],
      "text": "    export var a = 1 == 2",
    },
    "line": 15,
    "message": "Missing code language flag",
    "nodeType": "Program",
    "ruleId": "remark-lint-fenced-code-flag",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: adjacent.mdx 1`] = `
Array [
  Object {
    "column": 14,
    "endColumn": 15,
    "endLine": 1,
    "fix": Object {
      "range": Array [
        13,
        14,
      ],
      "text": "",
    },
    "line": 1,
    "message": "Delete \`·\`",
    "nodeType": null,
    "ruleId": "prettier/prettier",
    "severity": 2,
  },
]
`;

exports[`fixtures should match all snapshots: basic.mdx 1`] = `
Array [
  Object {
    "column": 18,
    "endColumn": 19,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        46,
        47,
      ],
      "text": " = ",
    },
    "line": 3,
    "message": "Replace \`=\` with \`·=·\`",
    "nodeType": null,
    "ruleId": "prettier/prettier",
    "severity": 2,
  },
]
`;

exports[`fixtures should match all snapshots: basic.tsx 1`] = `Array []`;

exports[`fixtures should match all snapshots: blank-lines.mdx 1`] = `
Array [
  Object {
    "column": 9,
    "endColumn": 12,
    "endLine": 1,
    "fix": Object {
      "range": Array [
        8,
        11,
      ],
      "text": " Box ",
    },
    "line": 1,
    "message": "Replace \`Box\` with \`·Box·\`",
    "nodeType": null,
    "ruleId": "prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 16,
    "endLine": 27,
    "fix": Object {
      "range": Array [
        548,
        563,
      ],
      "text": "# Hello, world!",
    },
    "line": 27,
    "message": "Don’t use multiple top level headings (3:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 1,
    "endColumn": 36,
    "endLine": 35,
    "fix": Object {
      "range": Array [
        651,
        686,
      ],
      "text": "# Here's a text gradient shortcode!",
    },
    "line": 35,
    "message": "Don’t use multiple top level headings (3:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: code-blocks.md 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        7,
        12,
      ],
      "text": "# abc",
    },
    "line": 3,
    "message": "Do not use headings with similar content (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        7,
        12,
      ],
      "text": "# abc",
    },
    "line": 3,
    "message": "Don’t use multiple top level headings (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
  Object {
    "column": 1,
    "endColumn": 4,
    "endLine": 17,
    "fix": Object {
      "range": Array [
        158,
        187,
      ],
      "text": "    export var a = 1 == 2",
    },
    "line": 15,
    "message": "Missing code language flag",
    "nodeType": "Program",
    "ruleId": "remark-lint-fenced-code-flag",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: details.mdx 1`] = `
Array [
  Object {
    "column": undefined,
    "fatal": true,
    "line": undefined,
    "message": "Parsing error: Cannot close \`paragraph\` (1:1-1:53): a different token (\`mdxJsxTextTag\`, 1:1-1:10) is open",
    "ruleId": null,
    "severity": 2,
  },
]
`;

exports[`fixtures should match all snapshots: jsx-in-list.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: leading-spaces.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: markdown.md 1`] = `Array []`;

exports[`fixtures should match all snapshots: no-unescaped-entities.mdx 1`] = `Array []`;

exports[`fixtures should match all snapshots: processor.mdx 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        7,
        12,
      ],
      "text": "# abc",
    },
    "line": 3,
    "message": "Do not use headings with similar content (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-duplicate-headings",
    "severity": 2,
  },
  Object {
    "column": 1,
    "endColumn": 6,
    "endLine": 3,
    "fix": Object {
      "range": Array [
        7,
        12,
      ],
      "text": "# abc",
    },
    "line": 3,
    "message": "Don’t use multiple top level headings (1:1)",
    "nodeType": "Program",
    "ruleId": "remark-lint-no-multiple-toplevel-headings",
    "severity": 1,
  },
]
`;

exports[`fixtures should match all snapshots: remark.md 1`] = `Array []`;

exports[`fixtures should match all snapshots: remark.mdx 1`] = `
Array [
  Object {
    "column": 2,
    "fatal": true,
    "line": undefined,
    "message": "Parsing error: Unexpected character \`!\` (U+0021) before name, expected a character that can start a name, such as a letter, \`$\`, or \`_\` (note: to create a comment in MDX, use \`{/* text */}\`)",
    "ruleId": null,
    "severity": 2,
  },
]
`;
